File: 1.R
# Title     : Operations on Iris Dataset
# Objective : Display summary and plots for all features
# Created by: @github/AP-Atul
# Created on: 12/10/20

#COLORS
# Pink: ff006a
# Red : f2374d
# Blue :37c3f2
# Orange : f25d37
# Greeen : 37f260

# read the dataset file
filePath <- "./datasets/iris.csv"
df <- read.table(file = filePath, header = FALSE, sep = ",")
names(df) <- c("Sepal.Length", "Sepal.Width", "Petal.Length", "Petal.Width", "Class")

# Feature
# OP: "Sepal.Length" "Sepal.Width"  "Petal.Length" "Petal.Width"  "Class"
names(df)

#Stats
summary(df)

# Histograms
# 1. Sepal.Length
hist(df$Sepal.Length, col = "#ff006a")

# 2. Sepal.Width
hist(df$Sepal.Width, col = "#f2374d")

# 3. Petal.Length
hist(df$Petal.Length, col = "#37c3f2")

# 4. Petal.Width
hist(df$Petal.Width, col = "#f25d37")

# Box plot
# 1. Sepal.Length
boxplot(df$Sepal.Length, col = "#ff006a")

# 2. Sepal.Width
boxplot(df$Sepal.Width, col = "#f2374d")

# 3. Petal.Length
boxplot(df$Petal.Length, col = "#37c3f2")

# 4. Petal.Width
boxplot(df$Petal.Width, col = "#f25d37")

# Mutiple BoxPlot
boxplot(df, col = "#37f260")

File: 1.py
import pandas as pd
from matplotlib import pyplot as plt

df = pd.read_csv("./iris.csv")
df.columns = ["SLength", "SWidth", "PLength", "PWidth", "Class"]

print(df.mean(), "\n")

print(df.median(), "\n")

print(df.describe(), "\n")

# all data points hist
df.hist()
plt.show()

# sepal length
df.iloc[:, 0].hist()
plt.show()

# sepal width
df.iloc[:, 1].hist()
plt.show()

# petal length
df.iloc[:, 2].hist()
plt.show()

# petal width
df.iloc[:, 3].hist()
plt.show()

# boxplot
df.boxplot()
plt.show()

# sepal length
df.boxplot(column=["SLength"])
plt.show()

# sepal width
df.boxplot(column=["SWidth"])
plt.show()

# petal length
df.boxplot(column=["PLength"])
plt.show()

# petal width
df.boxplot(column=["PWidth"])
plt.show()


